docs/contract/dataplanes-dataframes.md

```diff
@@ -0,0 +1,91 @@
+
+# Grafana data structure
+
+Grafana supports a variety of different data sources, each with its own data model. To make this possible, Grafana consolidates the query results from each of these data sources into one unified data structure called a **data frame**. The **data plane** is built on top of the data frame and includes additional information on what the data frame holds. 
+
+## Data frames overview  
+
+A data frame is a data structure that consolidates the query results from your data sources, providing a common container in Grafana. 
+
+:::caution
+
+Query responses are often more than one single data frame. 
+
+:::
+
+The data frame a column-oriented table (the _fields_) with metadata (the _frame_) attached. Since data frame columns can have labels attached to them (`key=value`, `key2=val`...), it can hold Prometheus like responses as well. 
+
+Each field in a data frame contains optional information about the values in the field, such as units, scaling, and so on. By adding field configurations to a data frame, Grafana can configure visualizations automatically. For example, you could configure Grafana to automatically set the unit provided by the data source.
```

Comment:
this seems very helpful for a datasource dev, would it be possible to show how to set the unit and/or other examples?